processor/logdedupprocessor/config.go

```diff
@@ -73,7 +76,19 @@ func (c Config) Validate() error {
 		return fmt.Errorf("timezone is invalid: %w", err)
 	}
 
-	return c.validateExcludeFields()
+	if err = c.validateExcludeFields(); err != nil {
+		return err
+	}
+
+	if err = c.validateIncludeFields(); err != nil {
+		return err
+	}
+
+	if len(c.ExcludeFields) > 0 && len(c.IncludeFields) > 0 {
+		return errors.New("cannot define both exclude_fields and include_fields")
+	}
```

Comment:
I'd do the `len` checks first so we don't have try to validate include or exclude fields which may then need to be removed.

```suggestion
	if len(c.ExcludeFields) > 0 && len(c.IncludeFields) > 0 {
		return errors.New("cannot define both exclude_fields and include_fields")
	}

	if err = c.validateExcludeFields(); err != nil {
		return err
	}

	if err = c.validateIncludeFields(); err != nil {
		return err
	}
```